File: src/main/java/com/Tom/SchoolRunner/Student.java
package com.Tom.SchoolRunner;

public class Student {
    String name;
    int english;
    int math;

    public Student(String name, int english, int math) {
        this.name = name;
        this.english = english;
        this.math = math;
    }

    public void print(){
        System.out.println(name+"\t"+english+"\t"+math+"\t"+ getPassOrFailed()+"\t" + getGrading()+"\t" + getAverage());
    }

    public int getAverage(){
        return  (english + math)/2;
    }

    public String getPassOrFailed(){
        String passOrFailed = "Failed";
        if(getAverage()>=60){
            passOrFailed = "Pass";
        }
        return  passOrFailed;
    }

    public char getGrading(){
        char grading = 'F';
        switch (getAverage()/10){
            case 10:
            case 9:
                grading = 'A';
                break;
            case 8:
                grading = 'B';
                break;
            case 7:
                grading = 'C';
                break;
            case 6:
                grading ='D';
//             default:
        }
        return  grading;
    }
}

File: src/main/kotlin/com/Tom/Student/Student.kt
package com.Tom.Student

import java.util.*

fun main(args: Array<String>) {
//    userInput()
    val s1 = student("Hank", 99, 88)
    s1.print()
}

private fun userInput() {
    val scanner = Scanner(System.`in`)
    print("請輸入姓名：")
    val name = scanner.next()
    print("請輸入英文成績：")
    val english = scanner.nextInt()
    print("請輸入數學成績：")
    val math = scanner.nextInt()
    val s1 = student(name, english, math)
    s1.print()
}

class student(val name : String, val english : Int, val math : Int){

    fun print(){
        println("$name\t$english\t$math\t${getAverage()}\t${grading()}\t${passOrFailed()}" )
    }

    fun getAverage() = (math + english) / 2

    fun grading() = when(getAverage()){
        in 90..100 -> 'A'
        in 80..89 -> 'B'
        in 70..79 -> 'C'
        in 60..69 -> 'D'
        else -> 'F'
    }

    fun passOrFailed() = if(getAverage()>=60) "Pass" else "Failed"

}